extensions/control-plane/provision/provision-blob/src/main/java/org/eclipse/edc/connector/provision/azure/blob/ObjectStorageConsumerResourceDefinitionGenerator.java

```diff
@@ -40,4 +38,9 @@ public class ObjectStorageConsumerResourceDefinitionGenerator implements Consume
         }
         return ObjectStorageResourceDefinition.Builder.newInstance().id(id).accountName(account).containerName(container).build();
     }
+
+    @Override
+    public boolean canGenerate(DataRequest dataRequest, Policy policy) {
+        return AzureBlobStoreSchema.TYPE.equals(dataRequest.getDestinationType());
```

Comment:
this is not the same condition as it was before. The fact that `getDestinationType()` is always `null` when `getDataDestination()` is also `null` is an implementation detail and you cannot assume that is always the case.

Reply:
`dataDestination` and `destinationType` cannot be null (as per validation in `DataRequest.Builder.build()` method)

Reply:
> dataDestination and destinationType cannot be null (as per validation in DataRequest.Builder.build() method)

what if that ever changes? suddenly you'd need to touch code (i.e. the res-def-gen) that has nothing to do with it. as @jimmarino said, defensive programming is not a bad trait.

Reply:
To be honest we never have been defensive, and that's good in my opinion, if a null object appears here it's an upstream problem, so having a NPE exception thrown is correct (is the same thing that is done by the `requireNotNull` BTW).
I wouldn't expect this to change, that's why we define interfaces, having a component that expects a null input is only added complexity, even more so here that we're talking about a `DataRequest` and a `Policy`, can't imagine a scenario where we accept them as null.

Reply:
Agreed, that's why I suggested `Objects.requireNotNull` and not to handle the null case - it doesn't pollute the code and provides a better indication of where the issue is.

Reply:
@jimmarino the point here is that having a null `DataRequest` or a null `Policy` would make the calling this call useless, we should validate them at a manager level and, if some of those are null we should fail the transfer process, but taking care of nulls in any component is definitely a bad practice

Reply:
@ndr-brt That definitely needs to be solved, but in a separate PR.

Reply:
The Generators already contain the Object.requireNotNull condition so that no method can receive null parameters